### DISCUSSION TITLE
coinbrew build Cbc successfull on master branch but fails on specific tag (2.10.11)

### DISCUSSION THREAD

**Discussion by mbrisson-ivadolabs** [coin-or/Cbc#649] [Q&A]
I want to use Cbc version 2.10.11 on Python official docker image python:3.12.3-bookworm on aarch64 (Apple M1).

I can build successfully Cbc master branch using coinbrew:

```
FROM python:3.12.3-bookworm as base

WORKDIR /opt/cbc

RUN wget https://raw.githubusercontent.com/coin-or/coinbrew/master/coinbrew
RUN chmod u+x coinbrew
RUN ./coinbrew fetch Cbc@master
RUN ./coinbrew build Cbc

ENV LD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV DYLD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV PATH="${PATH}:/opt/cbc/bin"
```

My issue is when I replace master with the tag 2.10.11. 

```
FROM python:3.12.3-bookworm as base

WORKDIR /opt/cbc

RUN wget https://raw.githubusercontent.com/coin-or/coinbrew/master/coinbrew
RUN chmod u+x coinbrew
RUN ./coinbrew fetch Cbc@2.10.11
RUN ./coinbrew build Cbc

ENV LD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV DYLD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV PATH="${PATH}:/opt/cbc/bin"
```

Now I can't build and  logs mentions **unable to guess system type**.  A
Any idea what is different in master that makes it work?  Thanks

Here is the complete trace

` => ERROR [base  7/12] RUN ./coinbrew build Cbc                                                                                                                               2.9s
------                                                                                                                                                                             
 > [base  7/12] RUN ./coinbrew build Cbc:                                                                                                                                          
0.092 Welcome to the COIN-OR fetch and build utility, version 2                                                                                                                    
0.092                                                                                                                                                                              
0.092 To use the legacy version of coinbrew, '$ wget https://raw.githubusercontent.com/coin-or/coinbrew/v1.0/coinbrew'                                                             
0.092 For help, run script with --help or see https://coin-or.github.io/coinbrew                                                                                                   
0.092 Please report issues at https://github.com/coin-or/coinbrew
0.092 Detailed build documentation at https://coin-or.github.io
0.092 
0.093 ./coinbrew: line 1634: warning: setlocale: LC_NUMERIC: cannot change locale (en_US.UTF-8): No such file or directory
0.096 Package will be built in /opt/cbc/build
0.096 
0.516 ################################################
0.516 ### Building version 2.10.11
0.516 ### with existing versions of dependencies.
0.516 ### Ctrl-c and run 'fetch' to switch versions
0.516 ### or to ensure correct dependencies
0.516 ################################################
0.516 
0.517 Caching configuration options...
0.518 
0.518 
0.518 Installation is done automatically following build and test of each project.
0.518 
0.519 Installation directory is writable.
0.519 
0.519 Package will be installed to /opt/cbc/dist 
0.519 
0.564 Disabling uninstalled packages
0.571 
0.571 ##################################################
0.571 ### Skipping ThirdParty/ASL (--enable-optional to build) 
0.571 ##################################################
0.571 
0.578 
0.578 ##################################################
0.578 ### Skipping ThirdParty/Glpk (--enable-optional to build) 
0.578 ##################################################
0.578 
0.584 
0.584 ##################################################
0.584 ### Skipping ThirdParty/Metis (--enable-optional to build) 
0.584 ##################################################
0.584 
0.591 
0.591 ##################################################
0.591 ### Skipping ThirdParty/Mumps (--enable-optional to build) 
0.591 ##################################################
0.591 
0.598 
0.598 ##################################################
0.598 ### Configuring Data/Sample 1.2.12 
0.598 ##################################################
0.598 
1.735 /opt/cbc/Data/Sample/config.guess: unable to guess system type
1.735 
1.735 This script, last modified 2007-05-17, has failed to recognize
1.735 the operating system you are using. It is advised that you
1.735 download the most up to date version of the config scripts from
1.735 
1.735   http://savannah.gnu.org/cgi-bin/viewcvs/*checkout*/config/config/config.guess
1.735 and
1.735   http://savannah.gnu.org/cgi-bin/viewcvs/*checkout*/config/config/config.sub
1.735 
1.735 If the version you run (/opt/cbc/Data/Sample/config.guess) is already up to date, please
1.735 send the following data and any information you think might be
1.735 pertinent to <config-patches@gnu.org> in order to provide the needed
1.735 information to handle your system.
1.735 
1.735 config.guess timestamp = 2007-05-17
1.735 
1.735 uname -m = aarch64
1.735 uname -r = 6.6.22-linuxkit
1.735 uname -s = Linux
1.735 uname -v = #1 SMP Fri Mar 29 12:21:27 UTC 2024
1.735 
1.735 /usr/bin/uname -p = unknown
1.735 /bin/uname -X     = 
1.735 
1.735 hostinfo               = 
1.735 /bin/universe          = 
1.735 /usr/bin/arch -k       = 
1.735 /bin/arch              = aarch64
1.735 /usr/bin/oslevel       = 
1.735 /usr/convex/getsysinfo = 
1.735 
1.735 UNAME_MACHINE = aarch64
1.735 UNAME_RELEASE = 6.6.22-linuxkit
1.735 UNAME_SYSTEM  = Linux
1.735 UNAME_VERSION = #1 SMP Fri Mar 29 12:21:27 UTC 2024
1.736 configure: error: cannot guess build type; you must specify one
1.759 
1.759 Configuration failed, re-running with output enabled
1.759 
1.803 checking for svnversion... yes
1.804 checking for egrep... grep -E
1.805 checking whether ln -s works... yes
1.805 checking for a BSD-compatible install... /usr/bin/install -c
1.807 checking whether build environment is sane... yes
2.831 checking for gawk... no
2.832 checking for mawk... mawk
2.833 checking whether make sets $(MAKE)... yes
2.844 checking whether to enable maintainer-specific portions of Makefiles... no
2.845 checking whether this is a VPATH configuration... yes
2.853 checking build system type... /opt/cbc/Data/Sample/config.guess: unable to guess system type
2.905 
2.905 This script, last modified 2007-05-17, has failed to recognize
2.905 the operating system you are using. It is advised that you
2.905 download the most up to date version of the config scripts from
2.905 
2.905   http://savannah.gnu.org/cgi-bin/viewcvs/*checkout*/config/config/config.guess
2.905 and
2.905   http://savannah.gnu.org/cgi-bin/viewcvs/*checkout*/config/config/config.sub
2.905 
2.905 If the version you run (/opt/cbc/Data/Sample/config.guess) is already up to date, please
2.905 send the following data and any information you think might be
2.905 pertinent to <config-patches@gnu.org> in order to provide the needed
2.905 information to handle your system.
2.905 
2.905 config.guess timestamp = 2007-05-17
2.905 
2.905 uname -m = aarch64
2.905 uname -r = 6.6.22-linuxkit
2.905 uname -s = Linux
2.905 uname -v = #1 SMP Fri Mar 29 12:21:27 UTC 2024
2.905 
2.905 /usr/bin/uname -p = unknown
2.905 /bin/uname -X     = 
2.905 
2.905 hostinfo               = 
2.905 /bin/universe          = 
2.905 /usr/bin/arch -k       = 
2.905 /bin/arch              = aarch64
2.905 /usr/bin/oslevel       = 
2.905 /usr/convex/getsysinfo = 
2.905 
2.905 UNAME_MACHINE = aarch64
2.905 UNAME_RELEASE = 6.6.22-linuxkit
2.905 UNAME_SYSTEM  = Linux
2.905 UNAME_VERSION = #1 SMP Fri Mar 29 12:21:27 UTC 2024
2.906 configure: error: cannot guess build type; you must specify one
2.928 
2.928 Here is the contents of config.log
2.928 
2.928 This file contains any messages produced by compilers while
2.928 running configure, to aid debugging if configure makes a mistake.
2.928 
2.928 It was created by DataSample configure 1.2.12, which was
2.928 generated by GNU Autoconf 2.59.  Invocation command line was
2.928 
2.928   $ /opt/cbc/Data/Sample/configure --disable-dependency-tracking --prefix=/opt/cbc/dist --with-coin-instdir=/opt/cbc/dist
2.928 
2.928 ## --------- ##
2.928 ## Platform. ##
2.928 ## --------- ##
2.928 
2.928 hostname = buildkitsandbox
2.928 uname -m = aarch64
2.928 uname -r = 6.6.22-linuxkit
2.928 uname -s = Linux
2.928 uname -v = #1 SMP Fri Mar 29 12:21:27 UTC 2024
2.928 
2.928 /usr/bin/uname -p = unknown
2.928 /bin/uname -X     = unknown
2.928 
2.928 /bin/arch              = aarch64
2.928 /usr/bin/arch -k       = unknown
2.928 /usr/convex/getsysinfo = unknown
2.928 hostinfo               = unknown
2.928 /bin/machine           = unknown
2.928 /usr/bin/oslevel       = unknown
2.928 /bin/universe          = unknown
2.928 
2.928 PATH: /usr/local/bin
2.928 PATH: /usr/local/sbin
2.928 PATH: /usr/local/bin
2.928 PATH: /usr/sbin
2.928 PATH: /usr/bin
2.928 PATH: /sbin
2.928 PATH: /bin
2.928 
2.928 
2.928 ## ----------- ##
2.928 ## Core tests. ##
2.928 ## ----------- ##
2.928 
2.928 configure:1335: checking for svnversion
2.928 configure:1351: found /usr/bin/svnversion
2.928 configure:1362: result: yes
2.928 configure:1394: checking for egrep
2.928 configure:1404: result: grep -E
2.928 configure:1409: checking whether ln -s works
2.928 configure:1413: result: yes
2.928 configure:1459: checking for a BSD-compatible install
2.928 configure:1514: result: /usr/bin/install -c
2.928 configure:1525: checking whether build environment is sane
2.928 configure:1568: result: yes
2.928 configure:1633: checking for gawk
2.928 configure:1662: result: no
2.928 configure:1633: checking for mawk
2.928 configure:1649: found /usr/bin/mawk
2.928 configure:1659: result: mawk
2.928 configure:1669: checking whether make sets $(MAKE)
2.928 configure:1689: result: yes
2.928 configure:1862: checking whether to enable maintainer-specific portions of Makefiles
2.928 configure:1871: result: no
2.928 configure:2261: checking whether this is a VPATH configuration
2.928 configure:2268: result: yes
2.928 configure:2277: checking build system type
2.928 configure:2286: error: cannot guess build type; you must specify one
2.928 
2.928 ## ---------------- ##
2.928 ## Cache variables. ##
2.928 ## ---------------- ##
2.928 
2.928 ac_cv_build_alias=
2.928 ac_cv_env_build_alias_set=
2.928 ac_cv_env_build_alias_value=
2.928 ac_cv_env_host_alias_set=
2.928 ac_cv_env_host_alias_value=
2.928 ac_cv_env_target_alias_set=
2.928 ac_cv_env_target_alias_value=
2.928 ac_cv_path_install='/usr/bin/install -c'
2.928 ac_cv_prog_AWK=mawk
2.928 ac_cv_prog_egrep='grep -E'
2.928 ac_cv_prog_have_svnversion=yes
2.928 ac_cv_prog_make_make_set=yes
2.928 
2.928 ## ----------------- ##
2.928 ## Output variables. ##
2.928 ## ----------------- ##
2.928 
2.928 ABSBUILDDIR=''
2.928 ACLOCAL='${SHELL} /opt/cbc/Data/Sample/missing --run aclocal-1.9'
2.928 ALWAYS_FALSE_FALSE=''
2.928 ALWAYS_FALSE_TRUE='#'
2.928 AMTAR='${SHELL} /opt/cbc/Data/Sample/missing --run tar'
2.928 AUTOCONF='${SHELL} /opt/cbc/Data/Sample/missing --run autoconf'
2.928 AUTOHEADER='${SHELL} /opt/cbc/Data/Sample/missing --run autoheader'
2.928 AUTOMAKE='${SHELL} /opt/cbc/Data/Sample/missing --run automake-1.9'
2.928 AUX_DIR=''
2.928 AWK='mawk'
2.928 BUILDTOOLSDIR=''
2.928 CYGPATH_W='echo'
2.928 DATASAMPLE_SVN_REV=''
2.928 DEFS=''
2.928 ECHO_C=''
2.928 ECHO_N='-n'
2.928 ECHO_T=''
2.928 EGREP='grep -E'
2.928 EXAMPLE_CLEAN_FILES=''
2.928 EXAMPLE_FILES=''
2.928 EXAMPLE_UNCOMPRESSED_FILES=''
2.928 HAVE_EXTERNALS_FALSE=''
2.928 HAVE_EXTERNALS_TRUE='#'
2.928 INSTALL_DATA='${INSTALL} -m 644'
2.928 INSTALL_PROGRAM='${INSTALL}'
2.928 INSTALL_SCRIPT='${INSTALL}'
2.928 INSTALL_STRIP_PROGRAM='${SHELL} $(install_sh) -c -s'
2.928 LIBEXT=''
2.928 LIBOBJS=''
2.928 LIBS=''
2.928 LIBTOOLM4=''
2.928 LN_S='ln -s'
2.928 LTLIBOBJS=''
2.928 MAINT='#'
2.928 MAINTAINER_MODE_FALSE=''
2.928 MAINTAINER_MODE_TRUE='#'
2.928 MAKEINFO='${SHELL} /opt/cbc/Data/Sample/missing --run makeinfo'
2.928 PACKAGE='datasample'
2.928 PACKAGE_BUGREPORT='https://projects.coin-or.org/BuildTools/'
2.928 PACKAGE_NAME='DataSample'
2.928 PACKAGE_STRING='DataSample 1.2.12'
2.928 PACKAGE_TARNAME='datasample'
2.928 PACKAGE_VERSION='1.2.12'
2.928 PATH_SEPARATOR=':'
2.928 SET_MAKE=''
2.928 SHELL='/bin/bash'
2.928 STRIP=''
2.928 VERSION='1.2.12'
2.928 VPATH_DISTCLEANFILES=''
2.928 abs_bin_dir='/opt/cbc/dist/bin'
2.928 abs_include_dir='/opt/cbc/dist/include'
2.928 abs_lib_dir='/opt/cbc/dist/lib'
2.928 abs_source_dir='/opt/cbc/Data/Sample'
2.928 ac_ct_STRIP=''
2.928 am__leading_dot='.'
2.928 am__tar='${AMTAR} chof - "$$tardir"'
2.928 am__untar='${AMTAR} xf -'
2.928 bindir='${exec_prefix}/bin'
2.928 build=''
2.928 build_alias=''
2.928 build_cpu=''
2.928 build_os=''
2.928 build_vendor=''
2.928 datadir='${prefix}/share'
2.928 exec_prefix='NONE'
2.928 have_autoconf=''
2.928 have_automake=''
2.928 have_svn=''
2.928 have_svnversion='yes'
2.928 host_alias=''
2.928 includedir='${prefix}/include'
2.928 infodir='${prefix}/info'
2.928 install_sh='/opt/cbc/Data/Sample/install-sh'
2.928 libdir='${exec_prefix}/lib'
2.928 libexecdir='${exec_prefix}/libexec'
2.928 localstatedir='${prefix}/var'
2.928 mandir='${prefix}/man'
2.928 mkdir_p='mkdir -p --'
2.928 oldincludedir='/usr/include'
2.928 prefix='/opt/cbc/dist'
2.928 program_transform_name='s,x,x,'
2.928 sbindir='${exec_prefix}/sbin'
2.928 sharedstatedir='${prefix}/com'
2.928 sysconfdir='${prefix}/etc'
2.928 target_alias=''
2.928 
2.928 ## ----------- ##
2.928 ## confdefs.h. ##
2.928 ## ----------- ##
2.928 
2.928 #define DATASAMPLE_VERSION "1.2.12"
2.928 #define DATASAMPLE_VERSION_MAJOR 1
2.928 #define DATASAMPLE_VERSION_MINOR 2
2.928 #define DATASAMPLE_VERSION_RELEASE 12
2.928 #define PACKAGE "datasample"
2.928 #define PACKAGE_BUGREPORT "https://projects.coin-or.org/BuildTools/"
2.928 #define PACKAGE_NAME "DataSample"
2.928 #define PACKAGE_STRING "DataSample 1.2.12"
2.928 #define PACKAGE_TARNAME "datasample"
2.928 #define PACKAGE_VERSION "1.2.12"
2.928 #define VERSION "1.2.12"
2.928 
2.928 configure: exit 1
2.929 
------
`

**Comment by svigerske** [comment]
The buildsystem on releases may be too old to recognize this system correctly. The one on master is based on a much more recent autotools, but has not been relased so far.
If staying with master is not an option, then try giving `--build=aarch64-pc-linux-gnu` or even `--build=x86_64-pc-linux-gnu` as argument to coinbrew.

**Reply by mbrisson-ivadolabs** [reply]
Hey tanks for the answer .. but no luck trying to pass the arch as argument.  I will wait for the next release.

**Reply by svigerske** [reply]
You don't want to wait for a release from master.

This worked for me (though on a different arm64 chip):
```
FROM python:3.12.3-bookworm as base

WORKDIR /opt/cbc

RUN wget https://raw.githubusercontent.com/coin-or/coinbrew/master/coinbrew
RUN chmod u+x coinbrew
RUN ./coinbrew fetch Cbc@2.10.11
RUN ./coinbrew build Cbc --build=x86_64-pc-linux-gnu

ENV LD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV DYLD_LIBRARY_PATH="/opt/cbc/dist/lib"
ENV PATH="${PATH}:/opt/cbc/bin"
```

**Comment by svigerske** [comment]
PS: More details at #587

**Comment by tkralphs** [comment]
Apparently, it works to do [this](https://github.com/coin-or/COIN-OR-OptimizationSuite/issues/6#issuecomment-2068570914). @svigerske, what do you think about updating these files. There would obviously be a version mismatch, but if it doesn't cause any problems and fixes these issues, maybe it's not a bad idea?

**Reply by svigerske** [reply]
Hmm, it reduces a bit the pressure to get new releases out. But ok, I can see that I include something into the run_autotools script of BuildTools stable/0.8 to have this happen.
